src/components/Pages/Admin/ToolbarEditor.vue

```diff
@@ -0,0 +1,461 @@
+                        if (a === '') array[i].splice(j, 1);
+                        else {
+                            array[i][j] = parseInt(a, 10);
+                            array[i][j] = this.icons.find(tool => tool.mode === array[i][j]);
+                        }
+                        this.$log.debug(`After a(${i},${j}) = ${array[i][j]}`);
+                    }
+                }
+            }
+
+            this.$log.debug('Array is', array);
+
+            return array;
+        },
+
+        saveToolbar(selectedToolbar) {
+            this.handleChange();
+
+            this.$log.debug(this.teacher);
+
+            const toolbars = this.teacher.toolbars;
+            let name;
+            let overwrite = false;
+
+            if (selectedToolbar.length > 0) {
+                // eslint-disable-next-line no-restricted-globals
+                if (confirm(`Overwrite selected toolbar? (${selectedToolbar[0]})`)) {
+                    name = selectedToolbar[0];
+                    overwrite = true;
+                }
+            // eslint-disable-next-line no-restricted-globals
+            } else name = prompt('Enter toolbar name');
+
+            const toolbar = { name, tools: this.content };
+            if (overwrite === true) {
+                // eslint-disable-next-line no-unused-vars
+                const idx = toolbars.findIndex(t => t.name === name);
+                toolbars[idx] = toolbar;
+
+                this.patch(
+                    [this.teacher.username, {
+                        toolbars,
+                    },
+                    ],
+                );
+            } else if (name) {
+                this.patch(
+                    [this.teacher.username, {
+                        toolbars: [...toolbars, toolbar],
+                    },
+                    ],
+                );
+            }
+
+            this.$log.debug('Teacher after', this.teacher);
+        },
+
+        addToolbar() {
+            this.handleChange();
+            this.$log.debug(this.teacher);
+
+            const toolbars = this.teacher.toolbars;
+
+            // eslint-disable-next-line no-restricted-globals
+            const name = prompt('Enter toolbar name');
+            if (name) {
+                const idx = toolbars.findIndex(t => t.name === name);
+
+                if (idx !== -1) {
+                    alert('Toolbar with such name exists. Cancelling overwrite.');
+                    return;
+                }
+
+                const toolbar = { name, tools: this.content };
+
+                this.patch(
+                    [this.teacher.username, {
+                        toolbars: [...toolbars, toolbar],
+                    },
+                    ],
+                );
+            }
+
+            this.$log.debug('Teacher after', this.teacher);
+        },
+
+        clearToolbar() {
+            this.lists = [[], [], [], [], [], [], [], [], [], [], [], []];
```

Comment:
This is potentially error prone as it is not clearly visible how many subarrays should be stored in the `lists` property. I'd rather propose something like this:
```javascript
this.lists = Array.from(new Array(12), _ => []);
```
This could be extracted to a helper function at the top of a file and used in the data initializer as well.